client/src/push-content.ts

```diff
@@ -47,14 +75,46 @@ export const getMessage = async (): Promise<string | undefined> => {
 
 export const pushContent = () => async () => {
   if (await canPush(getErrorDiagnosticsBySource())) {
-    await _pushContent(getRepo, getMessage, vscode.window.showInformationMessage, vscode.window.showErrorMessage)()
+    await _pushContent(getRepo, taggingDialog, getNewTag, getMessage, vscode.window.showInformationMessage, vscode.window.showErrorMessage)()
   }
 }
 
-export const _pushContent = (_getRepo: () => Repository, _getMessage: () => Thenable<string | undefined>, infoReporter: (msg: string) => Thenable<string | undefined>, errorReporter: (msg: string) => Thenable<string | undefined>) => async () => {
+export const _pushContent = (
+  _getRepo: () => Repository,
+  _taggingDialog: () => Thenable<string | undefined>,
+  _getTag: (repo: Repository, release: boolean) => string | undefined,
+  _getMessage: () => Thenable<string | undefined>,
+  infoReporter: (msg: string) => Thenable<string | undefined>,
+  errorReporter: (msg: string) => Thenable<string | undefined>
+) => async () => {
   const repo = _getRepo()
   const commitOptions: CommitOptions = { all: true }
 
+  const tagging = await _taggingDialog()
+  let tag: string | undefined
+  if (tagging !== 'No Tag') {
+    /* instanbul ignore next */
+    const tagMode = tagging === 'Release'
+    tag = _getTag(repo, tagMode)
+  }
+
+  if (tag !== undefined) {
```

Comment:
Since `_getTag` can return `undefined`, is it possible that the user requests a tag but the commit gets pushed without one since we don't check explicitly after L98 and error if `undefined` is returned there like in L107 below?

Reply:
Tagging and committing will remain separate.

Reply:
Hmmm...I don't see how that impacts my question here. Stated differently, I think it's possible given this code for a user to expect a tag but for one not to be applied because `_getTag` can return `undefined`. I think either the return signature of `_getTag` should be modified to not allow returning `undefined`, or an explicit check needs to happen after L98 to throw an error if the user clicks wanting a tag but (for whatever reason) `getTag` returns `undefined` and this branch isn't taken.